commons/com.b2international.commons/src/com/b2international/commons/config/ConfigurationFactory.java

```diff
@@ -184,14 +193,44 @@ private void resolveModuleConfigs(Configuration config) throws IOException {
 			final JsonNode moduleConfigNode = config.getModuleConfigurations().get(fieldName);
 			Object moduleConfig = null;
 			if (moduleConfigNode != null) {
+				replaceEnvironmentVariables(moduleConfigNode);
 				moduleConfig = factory.build(moduleConfigNode, fieldName);
 			} else {
 				moduleConfig = factory.build();
 			}
 			config.getConfigurationClassToNode().put(fieldType, moduleConfig);
 		}
 	}
-
+	
+	/**
+	 * Recursively replaces environment variables from a given node.
+	 * 
+	 * @param node
+	 */
+	private void replaceEnvironmentVariables(JsonNode node) {
+		final Iterator<Entry<String, JsonNode>> fields = node.fields();
+		while (fields.hasNext()) {
+			final Entry<String, JsonNode> field = fields.next();
+			
+			final String fieldName = field.getKey();
+			final String textValue = field.getValue().textValue();
+			if (!Strings.isNullOrEmpty(textValue)) {
+				final Matcher matcher = ENV_VAR_PATTERN.matcher(textValue); 
+				if (matcher.matches()) {
+					final String envVariableName = matcher.group(1);
+					if (System.getenv().containsKey(envVariableName)) {
+						final String envVariableValue = System.getenv(envVariableName);
+						((ObjectNode) node).put(fieldName, envVariableValue);
```

Comment:
⚠️ Potential type error!

Please write test cases for the following types to verify that this will indeed work correctly:
* Boolean
* Short
* Integer
* Float
* Long
* Float
* Double
* String